### ISSUE TITLE
Support multi-turn QnA Maker in VA

### ISSUE THREAD

**Issue by Dewain27** [microsoft/botframework-solutions#1888]
## User Story
As ... a bot developer
I want ... a want to be able to enable multi-turn QnA Maker knowledgebases in VA
so that ... I can enable multi-turn scenarios as needed.

## Acceptance Criteria 
*List your requirements.*
- [ ] Modify existing QnA logic to support either single-turn QnA or multi-turn QnA
- [ ] Should align to SDK implementation
- [ ] Validate with an example in FAQ,lu
- [ ] Should be able to exit "multi-turn" mode at will
- [ ] Should exit multi-turn mode when answer not found

## Resources
*Provide any screenshots or conversation mockups to support your request.*
*Learn how to mock up bot conversations with [.chat files](http://aka.ms/bfchatfile).*

## Notes
*Provide any additional information.*
Multi-turn QnA has the side affect of trying to keep control of the conversation flow after each turn. We need a good way to allow the user to exit multi-turn mode as needed.

Should reference this sample to get started: https://github.com/microsoft/BotBuilder-Samples/tree/master/experimental/qnamaker-prompting/csharp_dotnetcore

**Comment by cperson** [comment]
I notice that version 4.5.3 of botbuilder-ai doesn't support the _context_ property in QnAResults. I assume this VA integration relies on this [botbuilder-js commit](https://github.com/microsoft/botbuilder-js/commit/803b7f997abb47a567aac3bb36b8f045fea53d3b) and associated code to be published to npm. 

Is there a road map or other reliable way of tracking the potential release of such enhancements? @lauren-mills @trojenguri

**Comment by lauren-mills** [comment]
@cperson yes, we are dependent on support from the BF SDK to include this feature in our TypeScript implementation. As for the SDK roadmap, their GitHub projects and releases are the best place to track their upcoming changes.

**Comment by lauren-mills** [comment]
Need to check with SDK team on QnA Dialog and future updates

**Comment by lauren-mills** [comment]
Blocked by ongoing work in SDK to support multi-turn OOTB

**Comment by samaea** [comment]
My customer and I are about to use VA, primarily for QnAMaker (for now) - will it be difficult to leverage the experimental multi-turn code over to the VA (as in myself doing it)? https://github.com/microsoft/BotBuilder-Samples/tree/master/experimental/qnamaker-prompting/csharp_dotnetcore

Thanks!

**Comment by cperson** [comment]
@samaea I am working with TypeScript, but found it fairly simple to implement follow up prompts using the unsafe type 'any'. 

I didn't implement the complete multi-turn model because I didn't want the conversation flow limited to QnA dialog.

**Comment by darrenj** [comment]
Changes merged into new QnAMaker - @Batta32 tracking work to update typescript
